healthworkforce/features/utils.js

```diff
@@ -0,0 +1,327 @@
+}
+
+exports.ihrisMockServicePractitionerRole = async () => {
+  const name = 'PractitionerRole'
+
+  await verifyResourceDoesNotExistInFHIR('PractitionerRole', testPractitionerRoleBundle.entry[0].resource.identifier[0].value)
+
+  // create new practitionerRole for Dr Bob on ihris mock server
+  const response = await createResourceBundle(name, testPractitionerRoleBundle)
+
+  if (response.status != 201) throw Error(`Failed to create ${name} for testing`)
+
+  console.log('\n Created PractitionerRole resource (for Dr Bob) on mock ihris mock server')
+}
+
+exports.gofrMockServiceLocation = async () => {
+  const name = 'Location'
+
+  await verifyResourceDoesNotExistInFHIR('Location', testLocationBundle.entry[0].resource.identifier[0].value)
+
+  // create new location on gofr mock server
+  const response = await createResourceBundle(name, testLocationBundle)
+
+  if (response.status != 201) throw Error(`Failed to create ${name} for testing`)
+
+  console.log('\n Created Location resource (GoodHealth Clinic) on mock gofr mock server')
+}
+
+exports.gofrMockServiceOrganization = async () => {
+  const name = 'Organization'
+
+  await verifyResourceDoesNotExistInFHIR('Organization', testOrganizationBundle.entry[0].resource.identifier[0].value)
+
+  // create new organization on gofr mock server
+  const response = await createResourceBundle(name, testOrganizationBundle)
+
+  if (response.status != 201) throw Error(`Failed to create ${name} for testing`)
+
+  console.log('\n Created Organization resource (Clinical Lab) on mock gofr mock server...')
+}
+
+const getResource = (resource, identifierValue) => {
+  return axios({
+    url: `${OPENHIM_PROTOCOL}://${OPENHIM_API_HOSTNAME}:${OPENHIM_TRANSACTION_API_PORT}/hapi-fhir-jpaserver/fhir/${resource}?identifier:value=${identifierValue}`,
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Custom ${CUSTOM_TOKEN_ID}`,
+      'Cache-Control': 'no-cache'
+    }
+  })
+}
+
+const removeResource = (resource, identifierValue) => {
+  return axios({
+    url: `${OPENHIM_PROTOCOL}://${OPENHIM_API_HOSTNAME}:${OPENHIM_TRANSACTION_API_PORT}/hapi-fhir-jpaserver/fhir/${resource}?identifier:value=${identifierValue}`,
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Custom ${CUSTOM_TOKEN_ID}`,
+      'Cache-Control': 'no-cache'
+    }
+  })
+}
+
+const verifyResourceDoesNotExistInFHIR = async (resource, resourceId) => {
+  const response = await getResource(resource, resourceId)
+
+  if (response.data.total > 0) throw Error(
+    `Test aborted! ${resource} resource (identifier-value: ${resourceId}) used in test already exists and will be removed from the FHIR server`
+    )
+}
+
+const createResourceBundle = (resourceName, resourceBundle) => {
+  return axios({
+    url: `http://localhost:${MOCK_SERVER_PORT}/create-resource/${resourceName}`,
+    method: 'POST',
+    data: resourceBundle
```

Comment:
We should use the normal FHIR resource creation here as that is what we will be using with the actual (non-mock) services. This means we should send the resource itself instead of a bundle to our mock services. I think all you need to change is the post bodies to contain just the resource.